Assets/_BForBoss/_Weapons/Scripts/Weapons/MeleeWeapon.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Perigon.Entities;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+    public class MeleeWeapon
+    {
+        private const int BUFFER_SIZE = 10;
+            
+        private readonly IMeleeProperties _meleeProperties;
+
+        private Collider[] _enemyBuffer = new Collider[BUFFER_SIZE];
+        private float _currentCooldown = 0f;
+        private Action<bool> _onHitEntity;
+        public bool CanMelee => _currentCooldown == 0f;
```

Comment:
Because of floating point precision, would it be better for this to be 
_currentCoolDown <= 0f

I could be overthinking.

Reply:
Thats probably a good call